reader.go

```
@@ -0,0 +1,79 @@
+package sstable
+
+import (
+	"encoding/binary"
+	"fmt"
+	"io"
+	"log"
+)
+
+// Reader loads the index of an SSTable file into the memory.  If the
+// file doesn't have an index, it scans the file and builds the index.
+// After that, we can query key and values of a key.
+type Reader struct {
+	io.ReadSeeker
+	index map[string][]int64
+}
+
+func Open(r io.ReadSeeker) (*Reader, error) {
+	ss := &Reader{
+		ReadSeeker: r,
+		index:      make(map[string][]int64)}
+	if e := ss.loadIndex(); e != nil {
+		return nil, e
+	}
+	return ss, nil
+}
+
+func must(e error) {
+	if e != nil {
+		log.Fatalf("Fatal due to %v", e)
+	}
+}
+
+func (ss *Reader) hasIndex() (uint64, error) {
+	fileLength, e := ss.Seek(0, io.SeekEnd)
+	must(e)
```

I think we should not use `log.Fatalf` or `panic` when the program is not in a unrecoverable state. From the panic official documentation:

> Panic
> The usual way to report an error to a caller is to return an error as an extra return value. The canonical Read method is a well-known instance; it returns a byte count and an error. But what if the error is unrecoverable? Sometimes the program simply cannot continue.

https://golang.org/doc/effective_go.html#panic

Instead, I think we should just use error, this approach is very common in the golang standard packages. Here is a blog post explaining more: https://blog.golang.org/errors-are-values

In this specific case, `Seek` reported an error. Which only indicate there is an error reading the current file, the library has not reached to an unrecoverable state. (If it's unrecoverable, `Seek` don't need to return an error, it can simply panic).